ssg/variables.py

```diff
@@ -0,0 +1,415 @@
+    """
+    A class to represent profile variables.
+
+    Attributes:
+    -----------
+    id : str
+        The unique identifier for the profile.
+    product : str
+        The product associated with the profile.
+    variables : dict
+        A dictionary containing the variables for the profile.
+    """
+    def __init__(self, id, product, variables):
+        self.id = id
+        self.product = product
+        self.variables = variables
+
+
+def _load_product_yaml(content_dir: str, product: str) -> object:
+    """
+    Load the product YAML file and return its content as a Python object.
+
+    Args:
+        content_dir (str): The directory where the content is stored.
+        product (str): The name of the product.
+
+    Returns:
+        object: The loaded YAML content as a Python object.
+    """
+    file_yaml_path = product_yaml_path(content_dir, product)
+    return load_product_yaml(file_yaml_path)
+
+
+def _load_yaml_profile_file(file_path: str) -> dict:
+    """
+    Load the content of a YAML file intended to profiles definitions.
+
+    It is not necessary to process macros in this case.
+
+    Args:
+        file_path (str): The path to the YAML file.
+
+    Returns:
+        dict: The content of the YAML file as a dictionary.
+    """
+    with open(file_path, 'r') as file:
+        try:
+            return yaml.safe_load(file)
+        except yaml.YAMLError as e:
+            print(f"Error loading YAML profile file {file_path}: {e}")
+            return {}
+
+
+def _get_extended_profile_path(profiles_files: list, profile_name: str) -> str:
+    """
+    Retrieve the full path of a profile file from a list of profile file paths.
+
+    Args:
+        profiles_files (list of str): A list of file paths where profile files are located.
+        profile_name (str): The name of the profile to search for.
+
+    Returns:
+        str: The full path of the profile file if found, otherwise None.
+    """
+    profile_file = f"{profile_name}.profile"
+    profile_path = next((path for path in profiles_files if profile_file in path), None)
+    return profile_path
+
+
+def _process_profile_extension(profiles_files: list, profile_yaml: dict,
+                               profile_variables: dict, policies: dict) -> dict:
+    """
+    Processes the extension of a profile by recursively checking if the profile extends another
+    profile and updating the profile variables accordingly.
+
+    Args:
+        profiles_files (list): List of profile file paths.
+        profile_yaml (dict): The YAML content of the current profile.
+        profile_variables (dict): The variables already defined in the current profile.
+        policies (dict): The policies defined in the current profile.
+
+    Returns:
+        dict: The updated profile variables after processing the extended profile,
+              or the original profile variables if no extension is found.
+    """
+    extended_profile = profile_yaml.get("extends", None)
```

Comment:
None is not needed because the get() function returns None by default.
```suggestion
    extended_profile = profile_yaml.get("extends")
```

Reply:
Do you see any problem to make it explicit?

Reply:
I don't push it, actually I wouldn't notice this if Ruff did not put a warning there.
https://docs.astral.sh/ruff/rules/dict-get-with-none-default/
I am OK with keeping as it is if you don't want to change it.

Reply:
Ok. I will accept your suggestions.